src/App.tsx

```diff
@@ -34,62 +35,131 @@ export function getReorderedGoods(
   // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+      case SortType.NONE:
+        return 0;
+      default:
+        throw new Error('No sort type, write correct one');
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   return visibleGoods;
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+ type State = {
+   isReversed: boolean,
+   sortType: SortType,
+ };
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  sortByLength = () => (
+    this.setState({
+      sortType: SortType.LENGTH,
+    })
+  );
+
+  sortByAlphabet = () => (
+    this.setState({
+      sortType: SortType.ALPHABET,
+    })
+  );
```

Comment:
these functions shouldn't return anything, check correct braces
```suggestion
  sortByAlphabet = () => {
    this.setState({
      sortType: SortType.ALPHABET,
    });
  };
```

Reply:
Thx for comment,  i fixed braces and commited changes)